American Whitewater/Helpers/AWApiReachHelper.swift

```diff
@@ -0,0 +1,620 @@
+            }
+        }
+        
+    }
+    
+    /*
+     func fetchReachesByIds(reachIds: [String], callback: @escaping ReachCallback,
+                            callbackError: @escaping ReachErrorCallback)
+     
+     This updates reaches based on their IDs. This is great for updating the favorites, and groups of reaches
+    */
+    func fetchReachesByIds(reachIds: [String], callback: @escaping ReachCallback, callbackError: @escaping ReachErrorCallback) {
+        
+        let urlString = baseURL + "River/list/list/\(reachIds.joined(separator: ":"))/.json"
+        
+        Alamofire.request(urlString).responseJSON { (response) in
+            
+            switch response.result {
+                case .success(let value):
+
+                    var riversList: [AWReach] = []
+                                        
+                    let json = JSON(value)
+                    print("Total JSON Reaches from server: \(json.count)")
+                                        
+                    if let riversArray = json.array {
+                        for riverJSON in riversArray {
+                            let reach = AWReach(json: riverJSON)
+                            riversList.append(reach)
+                        }
+                    }
+                
+                    print("Processed \(riversList.count) favorite rivers.")
+                
+                    callback(riversList)
+
+                case .failure(let error):
+                    print("Failed trying to call: \(urlString)")
+                    print("Response: \(response)")
+                    print("Response Description: \(response.debugDescription)")
+                    print("HTTP Response: \(response.response.debugDescription)")
+
+                    callbackError(error)
+            }
+            
+        }
+        
+    }
+    
+    
+    func findOrNewReach(newReach: AWReach, context: NSManagedObjectContext) -> Reach {
+        let request = Reach.fetchRequest() as NSFetchRequest<Reach>
+        //print("nReach name: \(newReach.name ?? "na") ID: \(NSNumber(value: newReach.id ?? 0))")
+        guard let id = newReach.id else {
+            print("invalid id: \(newReach.id ?? -1)")
+            return Reach(context: context)            
+        }
+        
+        request.predicate = NSPredicate(format: "id == %i", id)
+                
+        guard let result = try? context.fetch(request), result.count > 0 else {
+            let reach = Reach(context: context)
+            reach.id = Int16(newReach.id ?? 0)
+            return reach
+        }
+
+        return result.first!
+    }
+
+    private func createOrUpdateReaches(newReaches: [AWReach], context: NSManagedObjectContext) {
+        context.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
+        
+        for newReach in newReaches {
+
+            let reach = findOrNewReach(newReach: newReach, context: context)
+
+            reach.name = newReach.name
```

Comment:
Minor: It would probably be cleaner to move this parsing and transform code somewhere else.